src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -565,7 +565,7 @@ private static size_t CurlReceiveHeadersCallback(IntPtr buffer, size_t size, siz
                             if (colonIndex > 0)
                             {
                                 string headerName = responseHeader.Substring(0, colonIndex);
-                                string headerValue = responseHeader.Substring(colonIndex + 1);
+                                string headerValue = responseHeader.Substring(colonIndex + 1).Trim();
```

Comment:
>is WinHttpHandler too aggressive here, or have I misunderstood?
To be more efficient, it might have been better for WinHttpHandler to specify the specific list of "whitespace" characters to trim away based on what the HTTP protocol would allow as "ignorable whitespace". Instead, I simply used `Trim()`. In practice, for example, there can't be things like <CR> or <LF> characters in the header values because it would never parse as valid HTTP protocol in the first place.

I think the safest thing to do right now is to keep CurlHandler the same as WinHttpHandler and use `Trim()'.  In parallel, I will consult with other HTTP experts here to determine what a better, more mimimal set of trim characters could be. Then, later, we can optimize the algorithm and perhaps re-factor into some Common code modules that both handlers could share.